rust/parse_ast/src/convert_ast/converter.rs

```diff
@@ -104,64 +104,76 @@ impl<'a> AstConverter<'a> {
       .copy_from_slice(&self.index_converter.convert(end, false).to_ne_bytes());
   }
 
-  pub(crate) fn convert_item_list<T, F>(
+  pub(crate) fn convert_item_list_with_out_state<T, F>(
```

Comment:
To keep the diff smaller and the mental overhead smaller, how about we call this again `convert_item_list`, but we call the other one e.g. `convert_item_list_internal`?